exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExporterWorker.java

```diff
@@ -105,16 +109,23 @@ boolean registerView(View view) {
         // Ignore views that are already registered.
         return true;
       } else {
-        // If we upload a view that has the same name with a registered view but with different
-        // attributes, Stackdriver client will throw an exception.
-        logger.log(
-            Level.WARNING,
-            "A different view with the same name is already registered: " + existing);
-        return false;
+        if (overrideExistingMetrics) {
+          deleteMetricDescriptor(view);
```

Comment:
can it fail without deleting the view? If so, what is the expectation?

Reply:
If this failed, the subsequent call to create MetricDescriptor would fail too. Both failures will be logged and silently skipped, and stats with the changed MetricDescriptor will not be exported. This is also the current behavior of exporter.

Reply:
I updated the logic here to fail fast if `deleteMetricDescriptor` failed, since if `deleteMetricDescriptor` failed, `createMetricDescriptor` will fail for sure.